File: temari_info_backend/src/models/InstituteModel.js
const mongoose = require('mongoose');

const instSchema = mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
    univ_id:{
        type: mongoose.Schema.Types.ObjectId,
        ref: "University",
        required:true
    },
    inst_name:{
        type: String,
        required: true
    },
    phone:{
        type: String,
        required: false,
    },
    email:{
        type: String,
        required: false
    },
    inst_description: {
        type: String,
        required: true
    },
    
    department_id:[
        {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Departments",
        required:true
        }
    ]    

});


module.exports = mongoose.model('Institute', instSchema);

File: temari_info_flutter/lib/institute/model/institute_model.dart
class Institute {
  final String? id;
  final String? univ_id;
  final String? inst_name;
  final String? phone;
  final String? email;
  final String inst_description;
  final List<Object>? department_id;

  Institute(
      {this.id,
      this.univ_id,
      this.inst_name,
      this.phone,
      this.email,
      required this.inst_description,
      this.department_id});

  factory Institute.fromJson(Map<String, dynamic> json) {
    return Institute(
        id: json['id'],
        univ_id: json['univ_id'],
        inst_name: json['inst_name'],
        phone: json['phone'],
        email: json['email'],
        department_id: json['department_id'],
        inst_description: json['inst_description']);
  }
}
